### PR TITLE
Add permission caching

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: fperms_iscore/permissions.py [hunk 1]
```diff
@@ -16,25 +18,56 @@
...
+
+def get_all_user_perm_slugs(request):
+    if hasattr(request.user, '_fperms_is_core_user_perm_slugs'):
+        return request.user._fperms_is_core_user_perm_slugs
+
+    cache_key = None
+    cache = get_cache()
+
+    if cache:
+        assert hasattr(request, 'session'), 'The cached permissions requres session middleware to be installed, ' \
```
Comment (Formulka) [inline fperms_iscore/permissions.py:54]: requires

File: fperms_iscore/permissions.py [hunk 2]
```diff
@@ -16,25 +18,56 @@
...
+                                            'and come before the message middleware in the MIDDLEWARE list'
+        cache_key = f'fperms_is_core-{request.user.pk}-{request.session.session_key}'
+        cache = caches[settings.IS_CORE_PERM_CACHE_NAME]
```
Comment (Formulka) [inline fperms_iscore/permissions.py:57]: don't you set it already on line 51? what is the difference? 🤔  also unlike the `get_cache` function, this doesn't check whether the setting exists

File: fperms_iscore/permissions.py [hunk 3]
```diff
@@ -16,25 +18,56 @@
...
+
+    perm_slugs = None
+
+    if cache:
+        perm_slugs = cache.get(cache_key)
```
Comment (Formulka) [inline fperms_iscore/permissions.py:62]: can't you move this inside the first `if cache:` and `perm_slugs = None` before it?

File: fperms_iscore/permissions.py (hunk 2 above)
Comment (matllubos) [reply fperms_iscore/permissions.py:57]: yes this is wrong :)

File: fperms_iscore/permissions.py (hunk 3 above)
Comment (matllubos) [reply fperms_iscore/permissions.py:62]: yes thanks

**Pull request merged**